core/conn/jdbc_type2/native/SqlInterface.cpp

```diff
@@ -246,10 +246,14 @@ SQLRETURN GetJDBCValues( SQLItemDesc_def *SQLItemDesc,
 		SQLItemDesc->signType      = TRUE;
 		break;
 	case SQLTYPECODE_DATETIME:
+		// Calculate the length based on YYYY/MM/DD HH:MM:SS.ffffff
+        SQLItemDesc->maxLen = 20 + 6;
 		switch (SQLItemDesc->datetimeCode)
 		{
 		case SQLDTCODE_DATE:
 			SQLItemDesc->ODBCDataType = SQL_TYPE_DATE;
+            // For DATE, update the maxLen since it only have YYYY/MM/DD
+            SQLItemDesc->maxLen = 10;
 			break;
 		case SQLDTCODE_TIME:
 			SQLItemDesc->ODBCDataType = SQL_TYPE_TIME;
```

Comment:
Maybe we need update all the maxLen for different datetime type. However I need write some unit test case to make it work well for each datetime type.

  So, for this PR, I only update SQLDTCODE_DATE for this issue .

  If you think we need update all the maxLen, we can keep this PR here , I will take some days to test it and update this PR later.